foundation/tests/org.jboss.tools.foundation.core.test/src/org/jboss/tools/foundation/core/test/ecf/URLTransportUtilTest.java

```diff
@@ -261,5 +234,21 @@ public void testCaching() {
 			fail(uee.getMessage()); // should never happen
 		}
 	}
+
+	private void executeWithHttpServer(ConsumerWithException consumer, int status, int delay) throws Exception {
+		Server server = startFakeHttpServer(status, delay);
+		try {
+			consumer.accept(server);
+		} finally {
+			try {
+				server.stop();
+			} catch (Exception e) {}
+		}		
+	}
+
+	@FunctionalInterface
+	interface ConsumerWithException {
```

Comment:
Do you really need this interface ? Can't you use java.util.function.Consumer ?

Reply:
@jeffmaury yes because Consumer does not allow throwing Exceptions which our consumers do via asserts etc. Do you see a better solution?

Reply:
assert throws unchecked exceptions so that should not be an issue but I think we have to deal with IOException and other kind of exceptions.

Reply:
yes, sorry, misstated, the checked exceptions like IOExceptions are the problem.